### PR TITLE
Add think-and-check

### PR DESCRIPTION
**Pull request by mfriesenhahn**
Changes proposed in this pull request:

- Add ResponseTypeThinkAndCheck component and documentation.
- Migrate the four posts with existing self-assessments (`/control-focus-with-tabindex`, `/labels-and-text-alternatives`, `/use-semantic-html`, and `/same-origin-policy`) to the think-and-check question type.
- Remove outdated assessment short codes and associated documentation in `/handbook/web-dev-components`.
- Minor fixes to `/handbook/self-assessment-components`.

### CODE REVIEW

**Comment by mfriesenhahn** [comment]
I'll rebase to master once `add-mc` is merged.

File: src/lib/components/ResponseThinkAndCheck/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    this.reset = this.reset.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren) {
+      this.prerenderedChildren = [];
+      this.option = [];
+
+      for (const child of this.children) {
+        if (child.getAttribute("data-role") === "rationale") {
```
Comment (samthor) [inline src/lib/components/ResponseThinkAndCheck/index.js]: Is there only ever going to be one of these?

Comment (mfriesenhahn) [reply src/lib/components/ResponseThinkAndCheck/index.js]: Right

**Comment by mfriesenhahn** [comment]
Hey, @kaycebasques, can you PTAL at the documentation and check the posts being migrated so I can land this beast? I'm determined to finish this today!😤

**Comment by mfriesenhahn** [comment]
Thanks, @samthor, for all your careful reviews and help! I couldn't have finished this without you!

**Pull request merged**